### PR TITLE
Like query algorithm support '_' match

### LINKED ISSUE #23029: Like query algorithm support '_' match
## Feature Request

**For English only**, other languages will not accept.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Is your feature request related to a problem?
The like query algorithm does not support '_' match
### Describe the feature you would like.
Like query algorithm support '_' match

### PR DESCRIPTION
**Pull request by gxxiong**
Ref #23029 .

Changes proposed in this pull request:
  -Like query algorithm support '_' match

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [X] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithm.java [hunk 1]
```diff
@@ -154,17 +154,13 @@ private String digest(final String plainValue) {
         StringBuilder result = new StringBuilder(plainValue.length());
         for (char each : plainValue.toCharArray()) {
             char maskedChar = getMaskedChar(each);
-            if ('%' == maskedChar) {
```
Comment (strongduanmu) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithm.java:157]: If the result of the mask operation is % or _, should we keep the original character?

File: docs/document/content/features/encrypt/limitations.cn.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ weight = 2
 +++
 
 - 需自行处理数据库中原始的存量数据；
+- 模糊查询支持%、_，暂不支持escape；
```
Comment (strongduanmu) [inline docs/document/content/features/encrypt/limitations.cn.md:7]: Please add a space between chinese characters and symbols(or english word).

File: features/encrypt/core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithmTest.java [hunk 3]
```diff
@@ -58,6 +58,8 @@ private Properties createProperties() {
     public void assertEncrypt() {
         assertThat(englishLikeEncryptAlgorithm.encrypt("1234567890%abcdefghijklmnopqrstuvwxyz%ABCDEFGHIJKLMNOPQRSTUVWXYZ",
                 mock(EncryptContext.class)), is("0145458981%`adedehihilmlmpqpqtutuxyxy%@ADEDEHIHILMLMPQPQTUTUXYXY"));
+        assertThat(englishLikeEncryptAlgorithm.encrypt("_1234__5678__",
+                mock(EncryptContext.class)), is("_0145__4589__"));
```
Comment (strongduanmu) [inline features/encrypt/core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithmTest.java:62]: Can you add integration test for `_` support?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithm.java (hunk 1 above)
Comment (gxxiong) [reply features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithm.java:157]: The getMaskedChar function returns % or _ directly

File: docs/document/content/features/encrypt/limitations.cn.md (hunk 2 above)
Comment (gxxiong) [reply docs/document/content/features/encrypt/limitations.cn.md:7]: ok

File: features/encrypt/core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithmTest.java (hunk 3 above)
Comment (gxxiong) [reply features/encrypt/core/src/test/java/org/apache/shardingsphere/encrypt/algorithm/like/CharDigestLikeEncryptAlgorithmTest.java:62]: ok

**Review by strongduanmu — APPROVED** [review]
Good job, merged.

**Pull request merged**